codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/server/protocol/JsonDeserializerGenerator.java

```diff
@@ -275,6 +277,27 @@ private GoWriter.Writable generateDeserializeIntegral(String ident, String castT
                 ));
     }
 
+    private GoWriter.Writable generateDeserializeFloat(String ident, String castTo, long min, long max) {
```

Comment:
This is for deserializing floating-points, the min/max should match accordingly. Note how you have to cast the _VALUE consts to (long) with what you have.